src/redux/books/actions.ts

```diff
@@ -1,21 +1,16 @@
 import { Dispatch } from 'redux';
+import { createTypes, completeTypes } from 'redux-recompose';
 import BookService from '@services/BookService';
 
-export const actions = {
-  GET_BOOKS: '@@BOOKS/GET_BOOKS',
-  GET_BOOKS_SUCCESS: '@@BOOKS/GET_BOOKS_SUCCESS',
-  GET_BOOKS_FAILURE: '@@BOOKS/GET_BOOKS_FAILURE'
-} as const;
+export const actions = createTypes(completeTypes(['GET_BOOKS']), '@@BOOKS');
 
 export const actionCreators = {
-  getBooks: () => async (dispatch: Dispatch) => {
-    dispatch({ type: actions.GET_BOOKS });
-    const response = await BookService.getBooks();
-    if (response.ok) {
-      dispatch({ type: actions.GET_BOOKS_SUCCESS, payload: response.data });
-    } else {
-      dispatch({ type: actions.GET_BOOKS_FAILURE, payload: response.problem });
-    }
+  getBooks: () => (dispatch: Dispatch) => {
+    dispatch({
+      type: actions.GET_BOOKS,
+      target: 'books',
+      service: BookService.getBooks
+    });
```

Comment:
When you don't add some behavior in the action you can return the object directly:
```
  getBooks: () => ({
      type: actions.GET_BOOKS,
      target: 'books',
      service: BookService.getBooks
    })
```
Because in the `Library` you use:
`dispatch(BookActions.getBooks())`
equal to
`dispatch({ type: actions.GET_BOOKS, target: 'books', service: BookService.getBooks })` (`getBooks`'s returned object)